roachtest: sqlsmith/setup=seed/setting=no-mutations failed

[(roachtest).sqlsmith/setup=seed/setting=no-mutations failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2071705&tab=buildLog) on [provisional_202007071743_v20.2.0-alpha.2@0b6e118bc1bcba4cfb4fc6c660153ec5be3989e8](https://github.com/cockroachdb/cockroach/commits/0b6e118bc1bcba4cfb4fc6c660153ec5be3989e8):

```
The test failed on branch=provisional_202007071743_v20.2.0-alpha.2, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/sqlsmith/setup=seed/setting=no-mutations/run_1
	sqlsmith.go:169,sqlsmith.go:199,test_runner.go:757: error: pq: internal error: runtime error: index out of range [0] with length 0
		stmt:
		SELECT
			tab_11251._jsonb AS col_26983,
			NULL AS col_26984,
			st_azimuth('0101000020E6100000000000000000F87F000000000000F87F':::GEOGRAPHY::GEOGRAPHY, '0101000020E6100000000000000000F03F000000000000F03F':::GEOGRAPHY::GEOGRAPHY)::FLOAT8
				AS col_26985
		FROM
			defaultdb.public.seed@[0] AS tab_11251
		WHERE
			tab_11251._bool;

```

<details><summary>More</summary><p>


Artifacts: [/sqlsmith/setup=seed/setting=no-mutations](https://teamcity.cockroachdb.com/viewLog.html?buildId=2071705&tab=artifacts#/sqlsmith/setup=seed/setting=no-mutations)
Related:
- #49251 roachtest: sqlsmith/setup=seed/setting=no-mutations failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Asqlsmith%2Fsetup%3Dseed%2Fsetting%3Dno-mutations.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/geo/geogfn/azimuth.go
+++ b/pkg/geo/geogfn/azimuth.go
@@ -49,6 +49,10 @@ func Azimuth(a *geo.Geography, b *geo.Geography) (*float64, error) {
 		return nil, errors.Newf("arguments must be POINT geometries")
 	}
 
+	if aPoint.Empty() || bPoint.Empty() {
+		return nil, errors.Newf("cannot call ST_Azimuth with POINT EMPTY")
+	}
+
 	if aPoint.X() == bPoint.X() && aPoint.Y() == bPoint.Y() {
 		return nil, nil
 	}

--- a/pkg/geo/geogfn/azimuth_test.go
+++ b/pkg/geo/geogfn/azimuth_test.go
@@ -107,4 +107,18 @@ func TestAzimuth(t *testing.T) {
 		_, err := Azimuth(mismatchingSRIDGeographyA, mismatchingSRIDGeographyB)
 		requireMismatchingSRIDError(t, err)
 	})
+
+	t.Run("errors on POINT EMPTY", func(t *testing.T) {
+		_, err := Azimuth(
+			geo.MustParseGeography("POINT EMPTY"),
+			geo.MustParseGeography("POINT(1 0)"),
+		)
+		require.EqualError(t, err, "cannot call ST_Azimuth with POINT EMPTY")
+
+		_, err = Azimuth(
+			geo.MustParseGeography("POINT(1 0)"),
+			geo.MustParseGeography("POINT EMPTY"),
+		)
+		require.EqualError(t, err, "cannot call ST_Azimuth with POINT EMPTY")
+	})
 }

--- a/pkg/geo/geomfn/azimuth.go
+++ b/pkg/geo/geomfn/azimuth.go
@@ -50,6 +50,10 @@ func Azimuth(a *geo.Geometry, b *geo.Geometry) (*float64, error) {
 		return nil, errors.Newf("arguments must be POINT geometries")
 	}
 
+	if aPoint.Empty() || bPoint.Empty() {
+		return nil, errors.Newf("cannot call ST_Azimuth with POINT EMPTY")
+	}
+
 	if aPoint.X() == bPoint.X() && aPoint.Y() == bPoint.Y() {
 		return nil, nil
 	}

--- a/pkg/geo/geomfn/azimuth_test.go
+++ b/pkg/geo/geomfn/azimuth_test.go
@@ -102,4 +102,18 @@ func TestAzimuth(t *testing.T) {
 		_, err := Azimuth(mismatchingSRIDGeometryA, mismatchingSRIDGeometryB)
 		requireMismatchingSRIDError(t, err)
 	})
+
+	t.Run("errors on POINT EMPTY", func(t *testing.T) {
+		_, err := Azimuth(
+			geo.MustParseGeometry("POINT EMPTY"),
+			geo.MustParseGeometry("POINT(1 0)"),
+		)
+		require.EqualError(t, err, "cannot call ST_Azimuth with POINT EMPTY")
+
+		_, err = Azimuth(
+			geo.MustParseGeometry("POINT(1 0)"),
+			geo.MustParseGeometry("POINT EMPTY"),
+		)
+		require.EqualError(t, err, "cannot call ST_Azimuth with POINT EMPTY")
+	})
 }

--- a/pkg/sql/logictest/testdata/logic_test/geospatial
+++ b/pkg/sql/logictest/testdata/logic_test/geospatial
@@ -1720,6 +1720,124 @@ FROM (VALUES
 2  1     1     SRID=4326;LINESTRING (0 0, 1 0, 1 1, 0 0)  NULL  SRID=4326;LINESTRING (0.1 0.1, 0.9 0.1, 0.9 0.9, 0.1 0.1)  NULL
 0  NULL  NULL  NULL                                       NULL  NULL                                                       NULL
 
+statement error argument must be LINESTRING geometries
+SELECT ST_MakePolygon('MULTIPOINT(0 0, 1 1)')
+
+statement error unknown signature: st_makepolygon\(string\)
+SELECT ST_MakePolygon('abc')
+
+statement error argument must be LINESTRING geometries
+SELECT ST_MakePolygon(
+    ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)'),
+    ARRAY[
+      ST_GeomFromText('MULTIPOINT(50 70, 70 70, 70 50, 50 50, 50 70)')
+    ]);
+
+statement error argument must be LINESTRING geometries
+SELECT ST_MakePolygon(
+    ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)'),
+    ARRAY[
+      'MULTIPOINT(50 70, 70 70, 70 50, 50 50, 50 70)'
+    ]);
+
+statement error argument must be LINESTRING geometries
+SELECT ST_MakePolygon(
+    ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)'),
+    ARRAY['abc']);
+
+# SRID set on shell but not interior rings
+statement error mixed SRIDs are not allowed
+SELECT ST_MakePolygon(
+    ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)', 4326),
+    ARRAY[
+      ST_GeomFromText('LINESTRING(50 70, 70 70, 70 50, 50 50, 50 70)')
+    ]);
+
+# SRIDs on shell and interior ring are unmatched
+statement error mixed SRIDs are not allowed
+SELECT ST_MakePolygon(
+    ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)', 4326),
+    ARRAY[
+      ST_GeomFromText('LINESTRING(50 70, 70 70, 70 50, 50 50, 50 70)', 3857)
+    ]);
+
+# SRIDs on interior rings are unmatched
+statement error mixed SRIDs are not allowed
+SELECT ST_MakePolygon(
+    ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)', 4326),
+    ARRAY[
+      ST_GeomFromText('LINESTRING(50 70, 70 70, 70 50, 50 50, 50 70)', 4326),
+      ST_GeomFromText('LINESTRING(60 60, 75 60, 75 45, 60 45, 60 60)', 3857)
+    ]);
+
+statement error shell must be closed
+SELECT ST_MakePolygon(ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 70)'));
+
+statement error shell must have at least 4 points
+SELECT ST_MakePolygon(ST_GeomFromText('LINESTRING(40 80, 80 80, 40 80)'));
+
+statement error shell must have at least 4 points
+SELECT ST_MakePolygon(ST_GeomFromText('LINESTRING(40 80, 40 80)'));
+
+statement error holes must be closed
+SELECT ST_MakePolygon(
+    ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)'),
+    ARRAY[
+      ST_GeomFromText('LINESTRING(50 70, 70 70, 70 50, 50 50, 50 60)')
+    ]);
+
+statement error holes must have at least 4 points
+SELECT ST_MakePolygon(
+    ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)'),
+    ARRAY[
+      ST_GeomFromText('LINESTRING(50 70, 70 70, 50 70)')
+    ]);
+
+statement error holes must have at least 4 points
+SELECT ST_MakePolygon(
+    ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)'),
+    ARRAY[
+      ST_GeomFromText('LINESTRING(50 70, 50 70)')
+    ]);
+
+query T
+SELECT ST_AsEWKT(ST_MakePolygon( ST_GeomFromText('LINESTRING(75 29,77 29,77 29, 75 29)')));
+----
+POLYGON ((75 29, 77 29, 77 29, 75 29))
+
+query T
+SELECT ST_AsEWKT(ST_MakePolygon( ST_GeomFromText('LINESTRING(75 29,77 29,77 29, 75 29)', 4326)));
+----
+SRID=4326;POLYGON ((75 29, 77 29, 77 29, 75 29))
+
+query T
+SELECT ST_AsEWKT(ST_MakePolygon(
+      ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)'),
+      ARRAY[
+        ST_GeomFromText('LINESTRING(50 70, 70 70, 70 50, 50 50, 50 70)')
+      ]));
+----
+POLYGON ((40 80, 80 80, 80 40, 40 40, 40 80), (50 70, 70 70, 70 50, 50 50, 50 70))
+
+query T
+SELECT ST_AsEWKT(ST_MakePolygon(
+      ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)'),
+      ARRAY[
+        ST_GeomFromText('LINESTRING(50 70, 70 70, 70 50, 50 50, 50 70)'),
+        ST_GeomFromText('LINESTRING(60 60, 75 60, 75 45, 60 45, 60 60)')
+      ]));
+----
+POLYGON ((40 80, 80 80, 80 40, 40 40, 40 80), (50 70, 70 70, 70 50, 50 50, 50 70), (60 60, 75 60, 75 45, 60 45, 60 60))
+
+query T
+SELECT ST_AsEWKT(ST_MakePolygon(
+      ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)', 4326),
+      ARRAY[
+        ST_GeomFromText('LINESTRING(50 70, 70 70, 70 50, 50 50, 50 70)', 4326)
+      ]));
+----
+SRID=4326;POLYGON ((40 80, 80 80, 80 40, 40 40, 40 80), (50 70, 70 70, 70 50, 50 50, 50 70))
+
 # Multi-Geometry related operations.
 query ITTT
 SELECT
@@ -2577,120 +2695,5 @@ SELECT ST_Azimuth(ST_Point(0, 0)::geography, ST_Point(0, 0)::geography)
 ----
 NULL
 
-statement error argument must be LINESTRING geometries
-SELECT ST_MakePolygon('MULTIPOINT(0 0, 1 1)')
-
-statement error unknown signature: st_makepolygon\(string\)
-SELECT ST_MakePolygon('abc')
-
-statement error argument must be LINESTRING geometries
-SELECT ST_MakePolygon( 
-    ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)'), 
-    ARRAY[
-      ST_GeomFromText('MULTIPOINT(50 70, 70 70, 70 50, 50 50, 50 70)')
-    ]);
-
-statement error argument must be LINESTRING geometries
-SELECT ST_MakePolygon( 
-    ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)'), 
-    ARRAY[
-      'MULTIPOINT(50 70, 70 70, 70 50, 50 50, 50 70)'
-    ]);
-
-statement error argument must be LINESTRING geometries
-SELECT ST_MakePolygon( 
-    ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)'), 
-    ARRAY['abc']);
-
-# SRID set on shell but not interior rings
-statement error mixed SRIDs are not allowed
-SELECT ST_MakePolygon( 
-    ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)', 4326), 
-    ARRAY[
-      ST_GeomFromText('LINESTRING(50 70, 70 70, 70 50, 50 50, 50 70)')
-    ]);
-
-# SRIDs on shell and interior ring are unmatched
-statement error mixed SRIDs are not allowed
-SELECT ST_MakePolygon( 
-    ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)', 4326), 
-    ARRAY[
-      ST_GeomFromText('LINESTRING(50 70, 70 70, 70 50, 50 50, 50 70)', 3857)
-    ]);
-
-# SRIDs on interior rings are unmatched
-statement error mixed SRIDs are not allowed
-SELECT ST_MakePolygon( 
-    ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)', 4326), 
-    ARRAY[
-      ST_GeomFromText('LINESTRING(50 70, 70 70, 70 50, 50 50, 50 70)', 4326), 
-      ST_GeomFromText('LINESTRING(60 60, 75 60, 75 45, 60 45, 60 60)', 3857)
-    ]);
-
-statement error shell must be closed
-SELECT ST_MakePolygon(ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 70)'));
-
-statement error shell must have at least 4 points
-SELECT ST_MakePolygon(ST_GeomFromText('LINESTRING(40 80, 80 80, 40 80)'));
-
-statement error shell must have at least 4 points
-SELECT ST_MakePolygon(ST_GeomFromText('LINESTRING(40 80, 40 80)'));
-
-statement error holes must be closed
-SELECT ST_MakePolygon(
-    ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)'),
-    ARRAY[
-      ST_GeomFromText('LINESTRING(50 70, 70 70, 70 50, 50 50, 50 60)')
-    ]);
-
-statement error holes must have at least 4 points
-SELECT ST_MakePolygon(
-    ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)'),
-    ARRAY[
-      ST_GeomFromText('LINESTRING(50 70, 70 70, 50 70)')
-    ]);
-
-statement error holes must have at least 4 points
-SELECT ST_MakePolygon(
-    ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)'),
-    ARRAY[
-      ST_GeomFromText('LINESTRING(50 70, 50 70)')
-    ]);
-
-query T 
-SELECT ST_AsEWKT(ST_MakePolygon( ST_GeomFromText('LINESTRING(75 29,77 29,77 29, 75 29)')));
-----
-POLYGON ((75 29, 77 29, 77 29, 75 29))
-
-query T 
-SELECT ST_AsEWKT(ST_MakePolygon( ST_GeomFromText('LINESTRING(75 29,77 29,77 29, 75 29)', 4326)));
-----
-SRID=4326;POLYGON ((75 29, 77 29, 77 29, 75 29))
-
-query T 
-SELECT ST_AsEWKT(ST_MakePolygon( 
-      ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)'), 
-      ARRAY[
-        ST_GeomFromText('LINESTRING(50 70, 70 70, 70 50, 50 50, 50 70)')
-      ]));
-----
-POLYGON ((40 80, 80 80, 80 40, 40 40, 40 80), (50 70, 70 70, 70 50, 50 50, 50 70))
-
-query T 
-SELECT ST_AsEWKT(ST_MakePolygon( 
-      ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)'), 
-      ARRAY[
-        ST_GeomFromText('LINESTRING(50 70, 70 70, 70 50, 50 50, 50 70)'), 
-        ST_GeomFromText('LINESTRING(60 60, 75 60, 75 45, 60 45, 60 60)')
-      ]));
-----
-POLYGON ((40 80, 80 80, 80 40, 40 40, 40 80), (50 70, 70 70, 70 50, 50 50, 50 70), (60 60, 75 60, 75 45, 60 45, 60 60))
-
-query T 
-SELECT ST_AsEWKT(ST_MakePolygon( 
-      ST_GeomFromText('LINESTRING(40 80, 80 80, 80 40, 40 40, 40 80)', 4326), 
-      ARRAY[
-        ST_GeomFromText('LINESTRING(50 70, 70 70, 70 50, 50 50, 50 70)', 4326)
-      ]));
-----
-SRID=4326;POLYGON ((40 80, 80 80, 80 40, 40 40, 40 80), (50 70, 70 70, 70 50, 50 50, 50 70))
+statement error POINT EMPTY
+SELECT st_azimuth('0101000020E6100000000000000000F87F000000000000F87F':::GEOGRAPHY::GEOGRAPHY, '0101000020E6100000000000000000F03F000000000000F03F':::GEOGRAPHY::GEOGRAPHY)::FLOAT8

